### PR TITLE
Remove ending punctuation from WordNodes

### PR DESCRIPTION
**Pull request by localjo**
This is a potential workaround for https://github.com/wooorm/nlcst/issues/4 but I'm almost sure that this solution is incorrect.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,13 +45,18 @@ function all(tree, file, config) {
...
         var isCorrect = true;
-        var word = toString(node);
 
-        if (includes(ignore, word)) {
+        if (ignoreLiteral && isLiteral(parent, index)) {
             return;
         }
 
-        if (ignoreLiteral && isLiteral(parent, index)) {
+        if (node.children[node.children.length - 1].type == 'PunctuationNode') {
```
Comment (wooorm) [inline index.js:53]: This fails on words with dots in them, like `e.g.`.

**Comment by wooorm** [comment]
I don’t really think that this temporary problem should be fixed, temporarily, in this project.

Especially like this, as it could break other plugins, and does break words like `e.g.` and abbreviations ending in a dot `mr.`.

Having it in `quality-docs` until a real fix is out is better, I feel. Is that OK, for now?

**Comment by localjo** [comment]
I'm not exactly sure where to put a fix like this in quality-docs. I guess I could add some kind of plugin for retext with workarounds that filter the tree. :-/

**Comment by wooorm** [comment]
Yeah, like that instead!

``` js
retext().use(function () {
  return function (node) {
    visit(node, 'WordNode', function () {
      if (/*...*/) { /*...*/ }
    });
  };
})
```

**Comment by localjo** [comment]
Cool. Doing this in quality-docs for now. https://github.com/SparkartGroupInc/quality-docs/commit/30b1e7cb31b57fac84814f5822a3ceb86eacf5cd

**Pull request closed without merging**